src/runtime_src/core/tools/xbmgmt2/SubCmdProgram.cpp

```diff
@@ -110,16 +113,19 @@ update_shell(unsigned int index, const std::string& flashType,
     throw xrt_core::error("Shell not specified");
 
   auto pri = std::make_unique<firmwareImage>(primary.c_str(), MCS_FIRMWARE_PRIMARY);
-  std::unique_ptr<firmwareImage> sec;
   if (pri->fail())
     throw xrt_core::error(boost::str(boost::format("Failed to read %s") % primary));
+
+  std::unique_ptr<firmwareImage> sec;
   if (!secondary.empty()) {
     sec = std::make_unique<firmwareImage>(secondary.c_str(), MCS_FIRMWARE_SECONDARY);
     if (sec->fail())
-      sec = nullptr;
+      throw xrt_core::error(boost::str(boost::format("Failed to read %s") % secondary));
```

Comment:
Not required
For readability it sometimes is best to split functionality between two lines, for example:
```
   if (sec->fail()) {
      const std::string errMsg = boost::str(boost::format("Failed to read %s") % secondary);
      throw xrt_core::error(errMsg);
  }
```